README.md

```diff
@@ -26,16 +26,21 @@ Fetches and creates versioned GitHub resources.
 * `insecure`: *Optional. Default `false`.* When set to `true`, concourse will allow
   insecure connection to your github API.
 
-* `release`: *Optional. Default `true`.* When set to `true`, `put` produces
-  release and `check` detects releases.  If `false`, `put` and `check` will ignore releases.
-  Note that releases must have semver compliant tags to be detected.
-
-* `pre_release`: *Optional. Default `false`.* When set to `true`, `put` produces
-  pre-release and `check` detects prereleases. If `false`, only non-prerelease releases
-  will be detected and published. Note that releases must have semver compliant
-  tags to be detected.
-  If `release` and `pre_release` are set to `true`, `put` produces
-  release and `check` detects prereleases and releases.
+* `release`: *Optional. Default `true`.* When set to `true`, `check` detects
+  final releases and `put` publishes final releases (as opposed to
+  pre-releases). If `false`, `check` will ignore final releases, and `put` will
+  publish pre-releases if `pre_release` is set to `true`
+  **note:** if both `release` and `pre_release` are set to `false`, `put` will
```

Comment:
kind of makes me wonder, shouldn't the resource disallow setting both `release` and `pre_release` to false, this might clear some clutter and allow for easier to grasp readme?

Reply:
The annoying thing is that `pre_release` is currently used in two different ways:

1. For `check` (to indicate "I want to fetch pre-release versions")
2. For `put` (to indicate "I want to create a pre-release)

In the `check` case, it makes sense for both `release` and `pre_release` to be specified - it's just `put` where it's a problem.

I think a better solution would be to rename `source.release` and `source.pre_release` to `source.include_final_releases` and `source.include_pre_releases`, and to add a new `params.pre_release` for `put` only.

There's an existing issue here https://github.com/concourse/github-release-resource/issues/40 for making the behaviour more obvious, but I don't want to introduce any breaking changes right now - so just a documentation update for the time being.

Reply:
fair enough